### PR TITLE
Autotest fixes

### PR DESCRIPTION
**Pull request by amilcarlucas**
Use exceptions in autotests
Many small autotest fixes

### CODE REVIEW

File: Tools/autotest/arduplane.py [hunk 1]
```diff
@@ -104,404 +104,421 @@ def init(self):
...
 
-        # fly the roll in manual
-        self.mavproxy.send('switch 6\n')
-        self.wait_mode('MANUAL')
-
-        while count > 0:
-            self.progress("Starting roll")
-            self.set_rc(1, 1000)
-            if not self.wait_roll(-150, accuracy=90):
-                self.set_rc(1, 1500)
```
Comment (karthikdesai) [inline Tools/autotest/arduplane.py]: @peterbarker On failure, the `set_rc(1,1500)` will be executed when an exception occurs. But this is more generalised. Any suggestions or is it already OK?

Comment (karthikdesai) [reply Tools/autotest/arduplane.py]: *On line 327

**Comment by peterbarker** [comment]
Not quite what I had in mind.

https://github.com/amilcarlucas/ardupilot/pull/3

That PR only covers Rover.

That PR un-indents a bunch of blocks in APMrover2 that this PR indents, and catches the exceptions one level higher.  If a test is just going to return false on catching an exception - may as well not bother!

I also put in a run_test function to wrap each distinct test - that was easy on Rover, as you can see.

**Comment by karthikdesai** [comment]
@peterbarker Alright. 
But is it possible to merge some of the stuff that has been corrected in this PR? The commits can be seperated/squashed, or is there any rework that is needed?
So it is easier to work on top of that :)

**Comment by khancyr** [comment]
About exception handling, I was thinking more like Peter did it . I think it is simplier to write

**Comment by karthikdesai** [comment]
@khancyr Okay working on it!
And regarding 
https://github.com/ArduPilot/ardupilot/commit/74e9d3dc00d26a3b3cead8c088eb34405c7e4484 commit, Plane and Quadplane have already these features. I just implemented the same for the other vehicles!

**Comment by peterbarker** [comment]
On Tue, 8 May 2018, Amilcar Lucas wrote:

> Can you review it ?

Really, really nice change, thanks!

  - regularised commit messages
  - added flake8 compliance
  - fixed Copter's auto test; please test final pushed version in future

Pushed it in.

**Comment by karthikdesai** [comment]
Thanks @peterbarker!

**Pull request closed without merging**